packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -94,19 +92,19 @@ class AppContext {
     return _values.putIfAbsent(type, () {
```

Comment:
Because of the closure here even if I capture _reentrantChecks in a local it does not get promoted.